rubin_scheduler/scheduler/surveys/field_survey.py

```diff
@@ -1,72 +1,324 @@
+    dec : `float`
+        The dec of the field to observe (degrees)
+    sequence : `list` [`str`]
+        The sequence of observations to take. (specify which filters to use).
+    nvisits : `dict` {`str`: `int`}
+        Dictionary of the number of visits in each filter.
+    exptime : `dict` {`str`: `float`}
+        Dictionary of the exposure time for visits in each filter.
+    nexp : dict` {`str`: `int`}
+        Dictionary of the number of exposures per visit in each filter.
+    ignore_obs : `list` [`str`] or None
+        Ignore observations with this string in the `scheduler_note`.
+        Will ignore observations which match subsets of the string, as well as
+        the entire string. Ignoring 'mysurvey23' will also ignore 'mysurvey2'.
+    accept_obs : `list` [`str`] or None
+        If match_obs is set, then ONLY observations which match these
+        strings in the `scheduler_note` will be counted for the survey.
+        A complete match must occur; substrings will not match.
+        (for obs_array too??)
+    survey_name : `str` or None.
+        The name to give this survey, for debugging and visualization purposes.
+        Also propagated to the 'target_name' in the observation.
+        The default None will construct a name based on the
+        RA/Dec of the field.
+    scheduler_note : `str` or None
+        The value to include in the scheduler note.
+        The scheduler note is for internal, scheduler, use for the purposes of
+        identifying observations to ignore or include for a survey or feature.
+    readtime : `float`
+        Readout time for computing approximate time of observing
+        the sequence. (seconds)
+    filter_change_time : `float`
+        Filter change time, on average. Used for computing approximate
+        time for the observing sequence. (seconds)
+    nside : `float` or None
+        Nside for computing survey basis functions and maps.
+        The default of None will use rubin_scheduler.utils.set_default_nside().
+    flush_pad : `float`
+        How long to hold observations in the queue after they
+        were expected to be completed (minutes).
+    """
 
-class FieldSurvey(DeepDrillingSurvey):
     def __init__(
         self,
         basis_functions,
         RA,
         dec,
-        sequence="rgizy",
-        nvis=[20, 10, 20, 26, 20],
-        exptime=30.0,
-        u_exptime=30.0,
-        nexp=2,
+        sequence="ugrizy",
+        nvisits=None,
+        exptimes=None,
+        nexps=None,
         ignore_obs=None,
-        survey_name="DD",
-        reward_value=None,
-        readtime=2.0,
+        accept_obs=None,
+        survey_name=None,
+        target_name=None,
+        scheduler_note=None,
+        readtime=2.4,
         filter_change_time=120.0,
         nside=None,
         flush_pad=30.0,
-        seed=42,
         detailers=None,
     ):
+        default_nvisits = {"u": 20, "g": 20, "r": 20, "i": 20, "z": 20, "y": 20}
```

Comment:
Mutables as kwarg defaults is frowned upon, but I think it's more readable and like them.

Reply:
Are you asking me to change this, or leave it as is? the kwarg is nvisits, but it will fall back to default_nvisits which is an attribute if it's not set.